### PR TITLE
Log and document "hidden" feature to turn off adaptive trace at run-time

### PR DESCRIPTION
**Pull request by marc-hb**
\+ other cleanups, comments and clarifications, see commit messages.

Zero functional change.

### CODE REVIEW

**Comment by marc-hb** [comment]
The Ethernet error `[  811.432681] kernel: e1000e 0000:00:1f.6 eno1: Hardware Error`  in https://sof-ci.01.org/sofpr/PR5154/build11502/devicetest/?model=TGLU_UP_HDA_ZEPHYR&testcase=check-suspend-resume-5 is cleared unrelated to this PR. Everything else is green.

The new message `INFO Adaptive filtering disabled by user` appears in the FW logs.

**Comment by marc-hb** [comment]
@kkarask could you find someone to help with this?
https://quickbuild.igk.intel.com/build/8260870
```
master > Check if PR is approved @ ptlpmssmain01:8842 
Composite step 'Check if PR is approved' failed due to unsatisfied success condition.
master > Check if PR is approved > Check if PR author is authorized @ ptlpmssmain01:8842 
java.lang.IllegalArgumentException: Text must not be null or empty
```

**Review by lyakh — APPROVED** [review]

File: src/lib/agent.c [hunk 1]
```diff
@@ -44,15 +44,19 @@ DECLARE_SOF_UUID("sa", sa_uuid, 0x5276b491, 0x5b64, 0x464e,
 
 DECLARE_TR_CTX(sa_tr, SOF_UUID(sa_uuid), LOG_LEVEL_INFO);
 
-#define perf_sa_trace(pcd, sa)					\
```
Comment (lgirdwood) [inline src/lib/agent.c:47]: Using a macro preserve ```__func__``` and ```__LINE__```. Can you check the output is correct as before with this change ?

**Comment by kkarask** [comment]
@marc-hb CI tests already passed.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**